JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1427,4 +1437,82 @@ Descriptives <- function(jaspResults, dataset, options) {
     return(TRUE)
   else
     return(FALSE)
-}
\ No newline at end of file
+}
+
+.XbarchartNoId <- function(jaspResults, dataset, options) {
+
+  ready <- (length(options$variables) > 1)
+  if (!ready)
+    return()
+
+  jaspResults[["XbarPlot"]] <- createJaspPlot(title = "X bar chart", width = 650, aspectRatio = 1)
+  jaspResults[["XbarPlot"]]$dependOn(c("Xbarchart", "variables"))
+  jaspResults[["XbarPlot"]]$position <- 11
+  XbarPlot <- jaspResults[["XbarPlot"]]
+
+  data1= na.omit(as.data.frame(dataset))
+  means <- rowMeans(data1)
+  subgroups <- 1:length(means)
+  data2 <- data.frame(subgroups = subgroups, means = means)
+  sixsigma <- qcc::qcc(data1, type = 'xbar', plot = F)
+  center <- sixsigma$center
+  sd1 <- sixsigma$std.dev
+  UCL <- max(sixsigma$limits)
+  LCL <- min(sixsigma$limits)
+
+  p <- ggplot2::ggplot(data2, ggplot2::aes(x = subgroups, y = means, group = 1))+ ggplot2::geom_line()+
+    ggplot2::ylab("Subgroup mean")+
+    ggplot2::xlab('Subgroup')+
+    ggplot2::theme_classic()+
+    ggplot2::scale_x_continuous(breaks = c(subgroups), limits = c(1,length(subgroups) + 1.5))+
+    ggplot2::geom_point(size = 2)+
+    ggplot2::geom_hline(yintercept = center, color = 'black')+
+    ggplot2::geom_hline(yintercept = c(center - sd1,center + sd1), linetype = "dashed", color = "red")+
+    ggplot2::geom_hline(yintercept = c(UCL,LCL), color = "red")+
+    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,center,label = sprintf("Mean = %g",round(center,3))))+
+    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,UCL,label = sprintf("UCL = %g",round(UCL,3))))+
+    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,LCL,label = sprintf("LCL = %g",round(LCL,3))))+
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y= ggplot2::element_text(size= ggplot2::rel(1.4)))+
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y= ggplot2::element_text(size= ggplot2::rel(1.4)))+
+    ggplot2::scale_y_continuous(limits = c(LCL,UCL), breaks = NULL)
```

Comment:
Okay, this is a good time to discuss ggplot2 semantics. If you compare your plot to the other JASP plots you'll probably notice that yours doesn't really fit in. Below some comments about the style and code, the same goes for the Rchart.

#### About the style

Instead of `ggplot2::theme_classic()` please add:
```r
... +
JASPgraphs::geom_rangeframe() +
JASPgraphs::themeJaspRaw()
```
`geom_rangeframe()` draws axis lines and `themeJaspRaw()` sets the theme.

#### About the code

Instead of `ggplot2::aes(x = subgroups, y = means, group = 1)` please add a group column to `data2`.

If you use `scale_y/x_continuous` it's cleanest if you also use it to specify the axis title. For example, you have now:
```r
... +
ggplot2::xlab('Subgroup') +
ggplot2::scale_x_continuous(breaks = c(subgroups), limits = c(1,length(subgroups) + 1.5)) +
...
```
which could be:
```r
ggplot2::scale_x_continuous(name = 'Subgroup', breaks = c(subgroups), limits = c(1,length(subgroups) + 1.5)) +
```
also, `c(subgroups)` is essentially `c(1:length(means))`, so you could just omit the `c()`.

next we have:
```r
    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,center,label = sprintf("Range = %g",round(center,3))))+
    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,UCL,label = sprintf("UCL = %g",round(UCL,3))))+
    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,LCL,label = sprintf("LCL= %g",round(LCL,3))))
```
which should just be one call to geom_label:
```r
# before the plot
dfLabel <- data.frame(
  x = length(subgroups) + 1.2,
  y = c(center, UCL, LCL),
  l = c(
    gettextf("Range = %g", round(center, 3)),
    gettextf("UCL = %g",   round(UCL, 3)),
    gettextf("LCL = %g",   round(LCL, 3))
  )
)
# make the plot
p <- ... +
  ggplot2::geom_label(data = dfLabel, mapping = ggplot2::aes(x = x, y = y, label = l), 
    inherit.aes = FALSE)
```
Setting `inherit.aes = FALSE` is key here, otherwise ggplot wil complain.